### DISCUSSION TITLE
how use relationship with model different connection

### DISCUSSION THREAD

**Discussion by septasite** [adonisjs/core#3661] [Help] [answered]
suppose I have 2 databases. 

- User -> mysql
- UserGame -> mssql

in User Model HasOne and in UserGame Model BelongsTo

in UserGame model i add 
- public static connection = 'mssql'
- public static table = 'UserInfo'

i am trying to get data via user model.
await User.query()
        .where('username', auth.user!.username)
        .preload('userGame')
        .firstOrFail()

but i have error 

> "select * from `UserInfo` where `UserName` in ('admin') - ER_NO_SUCH_TABLE: Table 'apigame.userinfo' doesn't exist"

- apigame is db in mysql. 
- usergame is db in mssql

he should read usergame.userinfo but is still read connection User Model that is apigame

if i get data only from UserGame model it works,

but if getting data from User Model and preload UserGame doesn't work, it still reads the first database which is mysql

**Comment by RomainLanz** [comment, marked as answer]
Hey @septasite! 👋🏻 

You cannot make relationships work across multiple connections.

**Comment by taler-hash** [comment]
this is impossible in adonis js but however in laravel is achieveable using with or eager loading thats the caveat in adonis js
